sky/clouds/oci.py

```diff
@@ -0,0 +1,429 @@
+                    use_spot=resources.use_spot,
+                    clouds='oci')
+                zones = [r for r in iter(regions) if r.name == region.name
+                        ][0].zones
+
+            if zones is not None:
+                zone = zones[0].name
+
+        return {
+            'instance_type': instance_type,
+            'custom_resources': custom_resources,
+            'region': region.name,
+            'cpus': cpus,
+            'memory': resources.memory,
+            'zone': zone,
+            'image': image_id,
+            'app_catalog_listing_id': listing_id,
+            'resource_version': res_ver,
+            'use_spot': resources.use_spot
+        }
+
+    def get_feasible_launchable_resources(self,
+                                          resources: 'resources_lib.Resources'):
+        if resources.instance_type is not None:
+            assert resources.is_launchable(), resources
+            resources = resources.copy(accelerators=None)
+            return ([resources], [])
+
+        def _make(instance_list):
+            resource_list = []
+            for instance_type in instance_list:
+                r = resources.copy(
+                    cloud=OCI(),
+                    instance_type=instance_type,
+                    # Setting this to None as OCI doesn't separately bill /
+                    # attach the accelerators.  Billed as part of the VM type.
+                    accelerators=None,
+                    cpus=None,
+                    memory=None,
+                )
+                resource_list.append(r)
+            return resource_list
+
+        # Currently, handle a filter on accelerators only.
+        accelerators = resources.accelerators
+        if accelerators is None:
+            # Return a default instance type with the given number of vCPUs.
+            default_instance_type = OCI.get_default_instance_type(
+                cpus=resources.cpus,
+                memory=resources.memory,
+                disk_tier=resources.disk_tier)
+
+            if default_instance_type is None:
+                return ([], [])
+            else:
+                return (_make([default_instance_type]), [])
+
+        assert len(accelerators) == 1, resources
+
+        acc, acc_count = list(accelerators.items())[0]
+        (instance_list, fuzzy_candidate_list
+        ) = service_catalog.get_instance_type_for_accelerator(
+            acc,
+            acc_count,
+            use_spot=resources.use_spot,
+            cpus=resources.cpus,
+            memory=resources.memory,
+            region=resources.region,
+            zone=resources.zone,
+            clouds='oci')
+        if instance_list is None:
+            return ([], fuzzy_candidate_list)
+
+        return (_make(instance_list), fuzzy_candidate_list)
+
+    @classmethod
+    def check_credentials(cls) -> Tuple[bool, Optional[str]]:
+        """Checks if the user has access credentials to this cloud."""
+
```

Comment:
We should check the oci package installation here, i.e. we should return False when the `oci` package is not installed.
```
try:
    import oci
except ImportError as e:
    return False, 'oci is not installed. Install it with `pip install oci`'
```